### PR TITLE
[Kogito-8799] Add Kubernetes security profile to Kaniko and builder pods

### PR DESCRIPTION
**Pull request by desmax74**
See:https://issues.redhat.com/browse/KOGITO-8799

**Description of the change:**

**Motivation for the change:**

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: platform/kaniko_cache.go [hunk 1]
```diff
@@ -94,6 +95,17 @@ func createKanikoCacheWarmerPod(ctx context.Context, client client.Client, platf
 							MountPath: builder.KanikoCacheDir,
 						},
 					},
+					SecurityContext: &corev1.SecurityContext{
+						AllowPrivilegeEscalation: util.Pbool(false),
+						Privileged:               util.Pbool(false),
+						RunAsNonRoot:             util.Pbool(true),
```
Comment (ricardozanini) [inline platform/kaniko_cache.go:101]: Can you please create an alias for these functions within our own `utils` package?

File: utils/kubernetes/container.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package kubernetes
+
+import (
+	"github.com/kiegroup/container-builder/util"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func SecurityDefaults() *corev1.SecurityContext {
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply utils/kubernetes/container.go:21]: Seccomp API was introduced in Kubernetes 1.19, which is OCP 4.7 that RH does not support anymore: https://access.redhat.com/solutions/4870701

We are fine. That said, we should restrict installations prior to kube 1.20 on OLM/Hub/Guides.

**Review by ricardozanini — COMMENTED** [review]
@desmax74 we are missing:

1) Upgrade the container-builder dependency to reflect the Kaniko pod
2) Revert the e2e to validate seccomp

And the alias I was referring to was the `util.Pbool`. We should not use container-builder utils all over.

We create an alias in our `utils` package for this funcs.

File: utils/kubernetes/container.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (spolti) [reply utils/kubernetes/container.go:21]: ok, nice :)

File: utils/kubernetes/container.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
```
Comment (spolti) [inline utils/kubernetes/container.go:1]: maybe reneme this file to security_context.go?

**Comment by spolti** [comment]
@MarianMacik as Tristan is on PTO can you please review?

File: utils/kubernetes/container.go (hunk 3 above)
Comment (desmax74) [reply utils/kubernetes/container.go:1]: ok

Comment (desmax74) [reply utils/kubernetes/container.go]: renamed

File: utils/kubernetes/security_context.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+// limitations under the License.
+package kubernetes
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/kiegroup/kogito-serverless-operator/utils"
+)
+
+// This is the security defualt to run without privileges
```
Comment (radtriste) [inline utils/kubernetes/security_context.go:22]:
```suggestion
// This is the security default to run without privileges
```

File: test/e2e/workflow_test.go [hunk 5]
```diff
@@ -210,7 +203,7 @@ var _ = Describe("Kogito Serverless Operator", Ordered, func() {
 			}, time.Minute, time.Second).Should(Succeed())
 		})
 
-		It("should successfully deploy the Greeting Workflow in prod mode and verify if it's running", func() {
+		/*It("should successfully deploy the Greeting Workflow in prod mode and verify if it's running", func() {
```
Comment (radtriste) [inline test/e2e/workflow_test.go:206]: why is this commented ?

**Review by MarianMacik — COMMENTED** [review]
Looks good. Just minor things + what Tristan have already found.

**Review by ricardozanini — COMMENTED** [review]
@desmax74 can we get back to this one? Is there anything else you need to address? Need help?

**Pull request closed without merging**